src/Ilios/CoreBundle/EventListener/RemoveOrphanedObjectives.php

```diff
@@ -0,0 +1,72 @@
+<?php
+namespace Ilios\CoreBundle\EventListener;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\Event\OnFlushEventArgs;
+use Doctrine\ORM\UnitOfWork;
+use Ilios\CoreBundle\Entity\CourseInterface;
+use Ilios\CoreBundle\Entity\Objective;
+use Ilios\CoreBundle\Entity\ProgramYearInterface;
+use Ilios\CoreBundle\Entity\SessionInterface;
+use Ilios\CoreBundle\Traits\IdentifiableEntityInterface;
+
+/**
+ * Doctrine event listener.
+ * When a ProgramYear, Course, or Session is deleted cleanup any newly orphaned objectives.
+ */
+class RemoveOrphanedObjectives
+{
+    /**
+     * @param OnFlushEventArgs $eventArgs
+     * @throws \Doctrine\ORM\ORMException
+     */
+    public function onFlush(OnFlushEventArgs $eventArgs)
+    {
+        $em = $eventArgs->getEntityManager();
+        $uow = $em->getUnitOfWork();
+
+        foreach ($uow->getScheduledEntityDeletions() as $entity) {
+            if ($entity instanceof ProgramYearInterface ||
+                $entity instanceof CourseInterface ||
+                $entity instanceof SessionInterface
+            ) {
+                $objectives = $entity->getObjectives();
+                foreach ($objectives as $objective) {
+                    $programYears = $objective->getProgramYears();
+                    $courses = $objective->getCourses();
+                    $sessions = $objective->getSessions();
+                    /** @var IdentifiableEntityInterface[] $allLinks */
+                    $allLinks = $programYears->toArray() + $courses->toArray() + $sessions->toArray();
```

Comment:
I'd use `array_merge()` here, the `+` operator produces inaccurate results here.

Reply:
That's what I get for trying new things! Fixed it.